boa_engine/src/vm/mod.rs

```diff
@@ -137,7 +137,7 @@ impl Context {
             opcode
         };
 
-        let _timer = Profiler::global().start_event(&format!("INST - {}", opcode.as_str()), "vm");
+        let _timer = Profiler::global().start_event(opcode.as_instruction_str(), "vm");
```

Comment:
Hmmm.... maybe it would be better to just conditionally compile this statement:
```rust
#[cfg(feature = "profiler"')]
let _timer = Profiler::global().start_event(opcode.as_instruction_str(), "vm");
```

Reply:
Yup, long term this looks a much cleaner way. We should probably conditionally compile all profiling code.

Reply:
> Hmmm.... maybe it would be better to just conditionally compile this statement:
> 
> ```rust
> #[cfg(feature = "profiler")]
> let _timer = Profiler::global().start_event(opcode.as_instruction_str(), "vm");
> ```

That’s a good point, I expected that line to be optimised out when profiler isnt set. It’s looking like that’s not the case?

Reply:
> That’s a good point, I expected that line to be optimised out when profiler isnt set. It’s looking like that’s not the case?

I expected that too. But it seems like the compiler cannot optimize away the format macro.
I created a compiler explorer example for this: https://rust.godbolt.org/z/35E8nfqca

When passing `&str` directly to the function, everything is optimized away.
When passing `&format!("")`, the function call is optimized away, but the formatting code stays.